src/pages/index.js

```diff
@@ -8,80 +8,172 @@ import { initialCards, cardData, options , cardSelector } from "../utils/constan
 imagePreviewPopup.setEventListeners();
 
-function handleImageClick(data) {
-  imagePreviewPopup.open(data);
-};
+const confirmationPopup = new PopupWithConfirmation({popupSelector: options.deletePopup});
+confirmationPopup.setEventListeners();
+
+const avatarPopup = new PopupWithForm("#avatar-modal", handleAvatarChange);
+avatarPopup.setEventListeners();
 
+const api = new Api({
+  url: "https://around-api.en.tripleten-services.com/v1/",
+  headers: {
+    authorization: "0d8f734d-caf1-45e3-b9d3-764b4099955a",
+    "Content-Type": "application/json"
+  }
+}); 
+
+let cardSection;
+
+const renderCard = (data) => {
+  const card = createCard(data);
+  cardSection.addItem(card);
+};
 
-function createCard(cardData) {
-  const card = new Card(cardData, cardSelector, handleImageClick)
-  return card.getView()
+Promise.all([api.getUserInfo(), api.getInitialCards()])
+  .then(([userData, cards]) => {
+    userInfo.setUserInfo(userData);
+    cardSection = new Section(
+      {
+        items: cards,
+        renderer: renderCard,
+      },
+      options.cardList
+    );
+    cardSection.renderItems();
+  })
+  .catch((err) => {
+    console.log(err);
+  });
+
+const userInfo = new UserInfo(
+  profileTitle, 
+  profileDescription,
+  profileAvatar
+  );
+
+function createCard(data) {
+  const cardElement = new Card(
+    {
+      data,
+      handleImageClick: (imgData) => {
+        imagePreviewPopup.open(imgData);
+      },
+      handleDeleteClick: () => {
+        confirmationPopup.open(() => {
+          confirmationPopup.renderLoading(true);
+          api.deleteCard(data._id)
+            .then(() => {
+              cardElement.removeCard();
+              confirmationPopup.close();
+            })
+            .catch((err) => console.log(`An error occured: ${err}`))
+            .finally(() => confirmationPopup.renderLoading(false));
+        });
+      },
+      handleLikeClick: () => {
+        if (cardElement.isLiked()) {
+          api
+            .removeLike(cardElement._cardId)
+            .then((res) => {
+              cardElement.setLikes(res.isLiked);
+            })
+            .catch((err) => {
+              console.log(`An error occured: ${err}`);
+            });
+        } else {
+          api
+            .addLike(cardElement._cardId)
+            .then((res) => {
+              cardElement.setLikes(res.isLiked);
+            })
+            .catch((err) => {
+              console.log(`An error occured: ${err}`);
+            });
+        }
+      },
+      userId: userInfo.getId(),
+    },
+    options.cardTemplate
+  );
+  return cardElement.getView();
 }
 
+
 function handleProfileEditSubmit(data) {
-  userInfo.setUserInfo(data.name, data.profession)
-  editProfilePopup.close()
+  api.updateProfile(data.name, data.profession)
+  .then((data) => {
+    userInfo.setUserInfo(data)
+  })
+  .catch((err) => {
+    console.log(`An error occured: ${err}`);
+  });
+  editProfilePopup.close();
```

Comment:
`  editProfilePopup.close();
`

You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well